flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionVertex.java

```diff
@@ -706,8 +706,10 @@ private Execution resetForNewExecutionInternal(final long timestamp, final long
 
 	public void tryAssignResource(LogicalSlot slot) {
 		if (!currentExecution.tryAssignResource(slot)) {
-			throw new IllegalStateException("Could not assign resource " + slot + " to current execution " +
+			final RuntimeException e = new IllegalStateException("Could not assign resource " + slot + " to current execution " +
```

Comment:
nit: I have never seen this style before. I think I would just declare it as `IllegalStateException` but it's not important.

Reply:
Will fix it. IllegalStateException looks better.

Reply:
Done.